sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -108,9 +108,12 @@ case class Exchange(
         // both cases, we must copy.
         true
       }
-    } else {
+    } else if (shuffleManager.isInstanceOf[HashShuffleManager]) {
```

Comment:
This check isn't fixing any bugs / correctness issues _yet_, but I thought that it might guard against hard-to-find future bugs if someone defines a new shuffle manager without updating the logic here.  It's extremely unlikely that anyone would do this, but in light of proposals like ParquetShuffleManager it seemed like the safest option.